src/pyclaw/sharpclaw/solver.py

```diff
@@ -244,6 +255,12 @@ def setup(self,solution):
         except KeyError:
             raise KeyError('Maximum CFL number is not provided.')
 
+        # Find which function evaluations need to be computed for 'DWRK' time integrators
+        # Lists self.fun_eval and self.funt_eval store the indices of non-zero columns of Shu-Osher arrays
+        if self.time_integrator == 'DWRK':
+            [self.fun_eval.append(i) for i in xrange(1,self.a.shape[0]) if any(abs(self.a[:,i]))>0]
```

Comment:
I think it would be clearer to write

    self.fun_eval = [i for i in xrange(1,self.a.shape[0]) if any(abs(self.a[:,i]))>0]

Also, `fun_eval` doesn't seem like an appropriate name for what this contains.